crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -167,10 +171,13 @@ pub struct AddressOwner {
 }
 
 #[allow(dead_code)]
-pub(crate) enum ObjectVersionKey {
+pub(crate) enum ObjectLookupKey {
     Latest,
-    LatestAt(u64),   // checkpoint_sequence_number
-    Historical(u64), // version
+    LatestAt(u64),
+    VersionAt {
+        version: u64,
+        checkpoint_viewed_at: Option<u64>,
```

Comment:
When do we end up using this mode of look-up? I would have thought we either do `VersionAt(u64)` (look-up by version) or `LatestAt(u64)` (look-up latest by checkpoint number) and can't think of a case where it's helpful to apply both limits.

Reply:
Gas, ObjectChange, and ObjectRead - I think when we load object at version, we still need to pass the `checkpoint_viewed_at` info so that the rendered object will have the context for fields selected on it